src/main/scala/cromwell/engine/backend/jes/JesAttributes.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.engine.backend.jes
+
+import java.net.URL
+
+import com.typesafe.config.{Config, ConfigFactory}
+import cromwell.util.ConfigUtil._
+
+case class JesAttributes(applicationName: String,
+                         project: String,
+                         executionBucket: String,
+                         endpointUrl: URL,
+                         authMode: GcsAuthMode,
+                         docker: Option[DockerAuthInformation])
+object JesAttributes {
+
+  def apply(): JesAttributes = {
+    val jesConf = ConfigFactory.load.getConfig("backend").getConfig("jes")
+    val refConf: Config = ConfigFactory.parseResources("jes.conf")
+
+    jesConf.checkValidWrapped(refConf, "Jes")
+    jesConf.warnNotRecognized(refConf, "Jes", "dockerAccount", "dockerToken")
```

Comment:
Yeah I agree it's not very clear, I'm not 100% happy with this but I couldn't find a perfect solution yet...
So `jes.conf` contains what a jes configuration should look like. However optional keys have to be commented, otherwise they will fail the `checkValid` if not present in the actual conf file.

`warnNotRecognized` looks for keys in the actual conf file that are not in the reference, and log a warning for them. However optional keys (like `dockerAccount` and `dockerToken`) should not trigger a warning although they are not in the reference (well they are but they're commented, hence not parsed). Which is why you can pass additional arguments to `warnNotRecognized` to say "do NOT issue a warning for those keys if you find them".